src/sentry/ingest/transaction_clusterer/tasks.py

```diff
@@ -62,8 +62,11 @@ def cluster_projects(projects: Sequence[Project]) -> None:
         if features.has("organizations:transaction-name-clusterer", project.organization):
             with sentry_sdk.start_span(op="txcluster_project") as span:
                 span.set_data("project_id", project.id)
+                tx_names = redis.get_transaction_names(project)
+                if len(list(tx_names)) < redis.MAX_SET_SIZE:
+                    return
                 clusterer = TreeClusterer(merge_threshold=MERGE_THRESHOLD)
-                clusterer.add_input(redis.get_transaction_names(project))
+                clusterer.add_input(tx_names)
```

Comment:
At this point, isn't the `tx_names` generator empty because `list(tx_names)` consumed it? That is, I would just assign `tx_names = list(redis.get_transaction_names(project))`.

Reply:
Good catch. I addressed this in [683423b](https://github.com/getsentry/sentry/pull/46503/commits/683423be5043ef65c4383cf4383f3f09ae924c83), by converting it to a list and then back to an iterator. I want to avoid making calls to the data storage if possible, and I assume this is a relatively cheap operation performance-wise.